src/csharp/Microsoft.Spark/Interop/Ipc/JvmThreadPool.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+
+namespace Microsoft.Spark.Interop.Ipc
+{
+    /// <summary>
+    /// This class corresponds to the ThreadPool we maintain on the JVM side. This class keeps
+    /// track of which .NET threads are still alive, and issues an rmThread command if a thread is
+    /// not.
+    /// </summary>
+    internal class JvmThreadPool
+    {
+        private readonly IJvmBridge _jvmBridge;
+        private readonly ConcurrentDictionary<int, Thread> _activeThreads;
+        private readonly Thread _activeThreadMonitor;
+
+        /// <summary>
+        /// Construct the JvmThreadPool.
+        /// </summary>
+        /// <param name="jvmBridge">The JvmBridge used to call JVM methods.</param>
+        /// <param name="threadGcInterval">The interval to GC finished threads.</param>
+        public JvmThreadPool(IJvmBridge jvmBridge, TimeSpan threadGcInterval)
+        {
+            _jvmBridge = jvmBridge;
+            _activeThreads = new ConcurrentDictionary<int, Thread>();
+            _activeThreadMonitor = new Thread(delegate ()
```

Comment:
does the timer need to be in a new thread?  Will you be able to achieve the same thing if you define the timer outside this thread object ?

Reply:
Changed.  It turns out that `Timer` does its own threading.